x/ccv/consumer/keeper/keeper.go

```diff
@@ -318,6 +343,27 @@ func (k Keeper) GetHeightValsetUpdateID(ctx sdk.Context, height uint64) uint64 {
 	return binary.BigEndian.Uint64(bz)
 }
 
+// IterateHeightValsetUpdateID iterates over the valset update ids recorded for a given block height
+func (k Keeper) IterateHeightValsetUpdateID(ctx sdk.Context, cb func(height uint64) bool) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte{types.HeightValsetUpdateIDBytePrefix})
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		height := binary.BigEndian.Uint64(iterator.Key()[1:])
+		if cb(height) {
+			break
+		}
+	}
+}
+
+// ClearHeightValsetUpdateIDs clears the valset update ids in the store
+func (k Keeper) ClearHeightValsetUpdateIDs(ctx sdk.Context) {
+	k.IterateHeightValsetUpdateID(ctx, func(height uint64) bool {
+		k.DeleteHeightValsetUpdateID(ctx, height)
```

Comment:
Is it okay to delete while iterating? Both `ClearPacketMaturityTimes` and `ClearOutstandingDowntimes` delete entries after the iteration.